src/optimizer/build_probe_side_optimizer.cpp

```diff
@@ -0,0 +1,147 @@
+#include "duckdb/optimizer/build_probe_side_optimizer.hpp"
+#include "duckdb/planner/operator/logical_join.hpp"
+#include "duckdb/execution/physical_plan_generator.hpp"
+#include "duckdb/planner/operator/logical_any_join.hpp"
+#include "duckdb/planner/operator/logical_get.hpp"
+#include "duckdb/planner/operator/logical_comparison_join.hpp"
+#include "duckdb/common/enums/join_type.hpp"
+
+namespace duckdb {
+
+static void GetRowidBindings(LogicalOperator &op, vector<ColumnBinding> &bindings) {
```

Comment:
Could we get a comment explaining why we're looking for rowid bindings, and why we shouldn't flip the joins these specific cases?